packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -107,6 +113,20 @@ module.exports = (
           }).then(() => {
             const markdownAST = remark.parse(markdownNode.internal.content)
 
+            if (pathPrefix) {
+              // Ensure relative links include `pathPrefix`
+              visit(markdownAST, `link`, node => {
+                if (
+                  node.url &&
+                  node.url.startsWith(`/`) &&
+                  !node.url.startsWith(`//`) &&
+                  !node.url.startsWith(pathPrefix)
```

Comment:
How about assuming any relative link in a markdown file should be appended with a `pathPrefix`? Otherwise, we can wrongfully skip adding the pathPrefix to the links starting with something similar to a pathPrefix value. For example, having a 'pathPrefix = "/web"' and a relative link `/website/details` we'll get a link with no `pathPrefix` added.

Reply:
Yes. Mb its better to skip `!node.url.startsWith(pathPrefix)` condition.
@KyleAMathews what You think?

Reply:
`gatsby-link` does not check if link `startsWith(pathPrefix)`, so looks like this code also should not

Reply:
@bodia-uz yeah — don't see a reason to add that check. E.g. what if there was a path named `/my-site/` but your prefix was also `/my-site/`. Then that path would mysteriously not be prefixed.

Reply:
PR Updated.